scratch-js/Sound.mjs

```diff
@@ -47,6 +110,513 @@ export default class Sound {
       this._markDone();
     }
 
-    this.audio.pause();
+    if (this.source) {
+      this.source.disconnect();
+      this.source = null;
+    }
+  }
+
+  downloadMyAudioBuffer() {
+    Sound.setupAudioContext();
```

Comment:
There are 5 different calls to `setupAudioContext`. Are they all necessary?

Reply:
They aren't all necessary. I put them there to prevent the context from ever *not* existing - but as an alternative, we could place a single `Sound.setupAudioContext` in the Project constructor, assuming that will always be run before anything else. That would make the code a little simpler, if potentially causing bugs.

Or, we could use a getter-style way, so the audio context is silently created if necessary:

```js
static get audioContext() {
  if (!this._audioContext) {
    this.setupAudioContext();
  }
  return this._audioContext;
}
```

That would probably be more reliable than anything else. Agreed? :P

Reply:
That sounds the most reliable to me! (though remember the `if (!this.audioContext)`check is only necessary in one place: either `setupAudioContext` or the getter)

Reply:
Right, thanks! Done: f212b8a107fc68b9bd5defa57f3cef98d65132a8